src/main/java/com/minecolonies/network/messages/BuildRequestMessage.java

```diff
@@ -80,20 +78,25 @@ public void toBytes(@NotNull ByteBuf buf)
         buf.writeInt(mode);
     }
 
-    @Nullable
     @Override
-    public IMessage onMessage(@NotNull BuildRequestMessage message, MessageContext ctx)
+    public void messageOnServerThread(final BuildRequestMessage message, final EntityPlayerMP player)
     {
         Colony colony = ColonyManager.getColony(message.colonyId);
         if (colony == null)
         {
-            return null;
+            return;
         }
 
         AbstractBuilding building = colony.getBuilding(message.buildingId);
         if (building == null)
         {
-            return null;
+            return;
+        }
+
+        //Verify player has permission to change this huts settings
+        //if (!colony.getPermissions().hasPermission(player, Permissions.Action.MANAGE_HUTS))
+        {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1199)